packages/extension-ui/src/components/AccountNamePasswordCreation.tsx

```diff
@@ -126,20 +135,34 @@ function AccountNamePasswordCreation({
             isFocused
             onChange={_onNameChange}
           />
-          <Password onChange={_onPasswordChange} />
-          <Dropdown
-            className={className}
-            label={t<string>('Show on network')}
-            onChange={_onChangeNetwork}
-            options={options}
-            value={genesisHash}
+          <Password
+            label={isDeriving ? t<string>('Set sub-account password') : undefined}
+            onChange={_onPasswordChange}
           />
+          {!isDeriving && (
+            <Dropdown
+              className={className}
+              label={t<string>('Show on network')}
+              onChange={_onChangeNetwork}
+              options={options}
+              value={genesisHash}
+            />
+          )}
         </div>
       </ScrollWrapper>
       <VerticalSpace />
       {onBackClick && buttonLabel && (
-        <ButtonArea footer={footer}>
-          <BackButton onClick={_onBackClick} />
+        <ButtonArea footer={!isDeriving && footer}>
+          {!!master && isDeriving ? (
```

Comment:
You don't need casting when using ternary operator.
```suggestion
          {master && isDeriving ? (
```